Shaders/RHI/TrianglesConstantBuffer.azsl

```diff
@@ -21,14 +21,32 @@
 
 #endif
 
+
+#if AZ_TRAIT_CONSTANT_BUFFER_ALIGNMENT <= 64
```

Comment:
Should this be 16?
I'm thinking that with no padding, the size of that struct is 80 bytes. 64 and 32 don't divide equally into 80 bytes, but 16 bytes does.

Reply:
if AZ_TRAIT_CONSTANT_BUFFER_ALIGNMENT == 64 it should be 3 and if AZ_TRAIT_CONSTANT_BUFFER_ALIGNMENT == 16 it should be 0

Reply:
I don't think an if-else statement is needed. They have solid mathematical relations. We can do:
#define AZ_TRAIT_CONSTANT_BUFFER_ALIGNMENT_IN_FLOAT4
(AZ_TRAIT_CONSTANT_BUFFER_ALIGNMENT / 16)
#define NUM_INSTANCE_INFO_PADDINGS
(AZ_TRAIT_CONSTANT_BUFFER_ALIGNMENT_IN_FLOAT4 + (5 / AZ_TRAIT_CONSTANT_BUFFER_ALIGNMENT_IN_FLOAT4 
 * AZ_TRAIT_CONSTANT_BUFFER_ALIGNMENT_IN_FLOAT4)  - 5)

In short, if the current data size is A (size in float4, here is 5), the alignment is B (size in float4), then the padding is
B + (A / B) * B - A

Reply:
m_padding0[0] would not compile, thus the excessive #if statements. 

This would have been more concise with if so: 

m_padding0[(AZ_TRAIT_CONSTANT_ALIGNMENT-(4*16+16)) % AZ_TRAIT_CONSTANT_ALIGNMENT)];

Reply:
Added more alignment sizes to handle the different cases.